antarest/study/repository.py

```diff
@@ -279,17 +280,23 @@ def delete(self, id_: str, *ids: str) -> None:
     def update_tags(self, study: Study, new_tags: t.Sequence[str]) -> None:
         """
         Updates the tags associated with a given study in the database,
-        replacing existing tags with new ones.
+        replacing existing tags with new ones (case-insensitive).
 
         Args:
             study: The pre-existing study to be updated with the new tags.
             new_tags: The new tags to be associated with the input study in the database.
         """
-        existing_tags = self.session.query(Tag).filter(Tag.label.in_(new_tags)).all()
-        new_labels = set(new_tags) - set([tag.label for tag in existing_tags])
-        study.tags = [Tag(label=tag) for tag in new_labels] + existing_tags
+        new_upper_tags = {tag.upper(): tag for tag in new_tags}
+        existing_tags = self.session.query(Tag).filter(func.upper(Tag.label).in_(new_upper_tags)).all()
+        for tag in existing_tags:
+            if tag.label.upper() in new_upper_tags:
+                new_upper_tags.pop(tag.label.upper())
+        study.tags = [Tag(label=tag) for tag in new_upper_tags.values()] + existing_tags
         self.session.merge(study)
         self.session.commit()
+        # Drop orphan tags
+        self.session.query(Tag).filter(~Tag.studies.any()).delete(synchronize_session=False)  # type: ignore
```

Comment:
Add a comment for developers, so that they realize that if a `tag` is not associated with any study it will be deleted. So if `tags` are to be associated with object other than `Study`, this code must be updated.